extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/sql/resources/SqlStatementResource.java

```diff
@@ -605,7 +626,10 @@ private Optional<SqlStatementResult> getStatementStatus(
               sqlStatementState,
               msqControllerTask.getQuerySpec().getDestination()
           ).orElse(null) : null,
-          null
+          null,
+          detail ? SqlStatementResourceHelper.getQueryStagesReport(msqTaskReportPayloadSupplier.get().orElse(null)) : null,
```

Comment:
Wont you call the overlord 3 times here ?

Reply:
@cryptoe Have raised a PR to get rid of the redundant calls: https://github.com/apache/druid/pull/16839

Thanks for pointing this out!